### ISSUE TITLE
broken ssl in Erlang 19

### ISSUE THREAD

**Issue by fooqri** [swelham/ms_luis#1]
Adding this in case others hit this issue. 

I was getting __{:error, :closed}__ response from a call to _MsLuis.get_intent(url)_. It looks related to a known issue and a fix is described at the top of the __httpoison__ readme page. They suggest using _ssl: [{:versions, [:'tlsv1.2']}]_  in the options of _HTTPoison.get()_

```
HTTPoison.get("https://example.com/", [], [ ssl: [{:versions, [:'tlsv1.2']}] ])
```
The _new()_ function in the _swelham/ivar_ module accepts options and the send function passes them through to _HTTPoison.request()_. The _MsLuis.get_intent()_ function does not seem to support use of opts in the request however, only accepting a query string. 

A fix that worked for me is to add opts to the __MsLuis.get_intent()__ function.
```
  def get_intent(query, opts \\ []) do
    query = URI.encode(query)

    with config     <- Application.get_env(:ms_luis, :config),
         {:ok, url} <- build_url(config, query)
    do
      Ivar.new(:get, url, opts)
      |> Ivar.send
      |> Ivar.unpack
      |> respond
    else
      err -> err
    end
  end
```

**Comment by swelham** [comment]
Ah yes, I believe this issue generally occurs when targeting a URL that is host by IIS 8. Would you be happy to submit a PR with your changes? If so, I would suggest having the ssl option hard coded into the request as this will always be needed.

**Comment by fooqri** [comment]
Happy to do a PR. I did just notice that the config for ms_luis has an ivar:http setting for ssl version. If I add this same config var to my phoenix project then everything works. Anyhow, this is going to be a case where the code is way faster to enhance than the decision for how best to enhance it. 

OPTIONS:
1. Leave code as is but update README recommending to add _:"tlsv1.2"_ to config. This seems a bit weird since it is config for a dependency:
```
config :ivar,
  http: [
    hackney: [ssl_options: [versions: [:'tlsv1.2']]]
  ]
```
2. Use the _:ms_luis_ config to declare the ssl_protocol. This has a code change to pick up and merge with opts before call to _ivar.new()_. Also update README to reflect option.
```
config :ms_luis, :config,
  url: "https://westus.api.cognitive.microsoft.com",
  app_key: "app_key",
  sub_key: "sub-key",
  ssl_protocol: :"tlsv1.2"
```
3. Hardcode it as the default for opts in _get_intent()_ or add it to opts at some point before calling _:ivar.new()_. 

I think 2 is probably most consistent with your current code. Let me know your preference.

**Comment by swelham** [comment]
I agree, option 2 is also my preference and having the `ssl_protocol` default to `tlsv1.2` makes sense as I don't believe this would need to be overridden in many situations.

**Comment by swelham** [comment]
Fixed by #2
